logdevice/common/RandomNodeSelector.cpp

```diff
@@ -71,7 +72,13 @@ size_t randomlySelectNodes(std::vector<node_index_t> candidates,
     return candidates.size();
   }
 
-  std::shuffle(candidates.begin(), candidates.end(), folly::ThreadLocalPRNG());
+  if (nodeOrderSeed) {
```

Comment:
Prefer explicitly checking for having a value instead. `nodeOrderSeed.has_value()`.